File: ecbs-os/frontend/src/lib/deploymentFieldWorkflowData.ts
import type { AnalysisSummaryRow } from "@/lib/analysisData";
import type { DocumentDataRow } from "@/lib/deploymentDocumentationData";

export type FieldEquipmentRow = {
  id: string;
  lastCommunicatedAt: string;
  location: string;
  name: string;
  rating: string;
  serialNumber: string;
  status: string;
  type: string;
};

export type FieldReadingRow = {
  delta: string;
  label: string;
  postValue: string;
  preValue: string;
  source: string;
  unit: string;
};

export type DeploymentFieldWorkflowData = {
  clientName: string;
  deploymentId: string;
  documentRows: DocumentDataRow[];
  equipmentRows: FieldEquipmentRow[];
  message: string;
  postReadingRows: FieldReadingRow[];
  preReadingRows: FieldReadingRow[];
  projectName: string;
  siteName: string;
  siteRows: AnalysisSummaryRow[];
  state: string;
  status: string;
  summaryRows: AnalysisSummaryRow[];
  updatedAt: string;
};

File: ecbs-os/backend/src/ECBS.Application/ScreenData/DeploymentFieldWorkflowData.cs
namespace ECBS.Application.ScreenData;

public sealed record DeploymentFieldWorkflowData(
    IReadOnlyList<AnalysisSummaryRow> SummaryRows,
    IReadOnlyList<AnalysisSummaryRow> SiteRows,
    IReadOnlyList<FieldEquipmentRow> EquipmentRows,
    IReadOnlyList<FieldReadingRow> PreReadingRows,
    IReadOnlyList<FieldReadingRow> PostReadingRows,
    IReadOnlyList<DocumentDataRow> DocumentRows,
    string ClientName,
    string DeploymentId,
    string Message,
    string ProjectName,
    string SiteName,
    string State,
    string Status,
    string UpdatedAt);

public sealed record FieldEquipmentRow(
    string Id,
    string Name,
    string Type,
    string SerialNumber,
    string Location,
    string Rating,
    string Status,
    string LastCommunicatedAt);

public sealed record FieldReadingRow(
    string Label,
    string PreValue,
    string PostValue,
    string Delta,
    string Unit,
    string Source);

